<!DOCTYPE html>
<!--[if lt IE 7]>      <html class="no-js lt-ie9 lt-ie8 lt-ie7"> <![endif]-->
<!--[if IE 7]>         <html class="no-js lt-ie9 lt-ie8"> <![endif]-->
<!--[if IE 8]>         <html class="no-js lt-ie9"> <![endif]-->
<!--[if gt IE 8]><!-->
<html class="no-js">
<!--<![endif]-->

<head>

	<meta charset="utf-8">
	<meta http-equiv="X-UA-Compatible" content="IE=edge">

	<title>About | VizRecon</title>

	<meta name="viewport" content="width=device-width, initial-scale=1">


	<meta name="description" content="" />
	<meta name="keywords" content="" />
	<meta name="author" content="" />


	<!-- google font -->
	<link href="https://fonts.googleapis.com/css?family=Montserrat|Open+Sans" rel="stylesheet">

	<!-- Place favicon.ico and apple-touch-icon.png in the root directory -->
	<link rel="shortcut icon" href="favicon.ico">

	<!-- Animate.css -->
	<link rel="stylesheet" href="css/animate.css">


	<!-- Theme Style -->
	<link rel="stylesheet" href="css/style.css">


	<!-- Modernizr JS -->
	<script src="js/modernizr-2.6.2.min.js"></script>

	<!-- FOR IE9 below -->
	<!--[if lt IE 9]>
	<script src="js/respond.min.js"></script>
	<![endif]-->

</head>

<body>
	<!-- Preloader -->
	<div id="loading">
		<div class="loader">
				<center><img style="-webkit-touch-callout: none;-webkit-user-select: none;-khtml-user-select: none;-moz-user-select: none;-ms-user-select: none;user-select: none;" src="images/a-loader.gif"></center>
		</div>
	</div>
	<!--End of Preloader -->
	<!-- Title bar -->
	<div id="vr-page">
		<header id="vr-header" class="">
			<div class="container">
				<div class="row">
					<div class="header-logo">
						<a href="index">
							VizRecon
						</a>
					</div>
					<div class="header-menu">

						<a class="dropdown-toggle" href="#" id="dropdownMenu1" data-toggle="dropdown">
							About
						</a>
						<ul class="dropdown-menu" aria-labelledby="dropdownMenu1">
							<li>
								<a href="aboutModel">About the Model</a>
							</li>
							<li>
								<a href="seniorDesign">Senior Design</a>
							</li>
							<li>
								<a href="people">People</a>
							</li>
						</ul>

						<a href="download">
							Downloads
						</a>
						<button type="button" disabled class="grey">log in</button>
						<button type="button" class="green" onclick="window.location.href='app'">start</button>
					</div>
				</div>
			</div>
		</header>

		<div id="section-about" class="section-about d-flex justify-content-center flex-column">
			<div class="container">
				<div class="row">
					<div class="d-flex col-md-12 col-sm-12">
						<div class="section-0-content col-md-8 col-sm-12">
							<h2 class="section-title">
								About the Model
							</h2>
							<h3 class="section-subtitle">
								Learn about how we constructed this 3D structural visualization of an E. coli cell and how we set it on display
							</h3>
						</div>
					</div>
				</div>
			</div>
		</div>
		<div class="clearfix"></div>

		<div id="section-1" class="section-1">
			<div class="container logos">
				<div class="row">
					<div class="col-md-12">
						<h3 class="section-1-title">
							Abstract
						</h3>
						<p>
							No complete, three-dimensional reconstruction of a prokaryote-scale cell exists to represent proteomic, metabolomic, and
							genomic data in an accessible manner. With the availability of experimental and computational information on the structures,
							stoichiometry, and localization of cellular components of
							<i>Escherichia coli</i>, along with the development of visualization tools such as cellPACK and chromosome simulation
							models, we have the “ingredients” we need to put together a model for the well-studied organism. This project integrates
							this data into an interactive, three-dimensional structural biology model of an
							<i>E. coli</i> cell for modern web browsers. The cell model contains still structural representations of the proteins,
							protein complexes, cell membrane, and chromosome, and is built on a pipeline that can produce a similar model from
							different data. It can be used as a stepping stone for organizing the information available on
							<i>E. coli</i> and as a stepping stone to develop models for other organisms at the prokaryotic and eukaryotic scales.
						</p>
					</div>
					<div>
						<center>
							<img src="images/FINALArtboard 1@2x.png" class="img-responsive" style="max-height: 450px">
						</center>
					</div>

					<div class="col-md-12 col-sm-12">
						<h3 class="section-1-title">
							Constructing the Model
						</h3>
						<div>
							<p class="pull-right">
								<a id="expand" button class="section-1-title" type="button" data-toggle="collapse" data-target=".collapse" aria-expanded="false"
								 aria-controls="heading1 heading2 heading3 heading4 heading5 heading6 heading7 heading15" style="cursor: pointer;"
								 onclick="if(document.getElementById('expand').innerHTML=='+ Expand all'){document.getElementById('expand').innerHTML='- Collapse all';}else{document.getElementById('expand').innerHTML='+ Expand all';}">+ Expand all</a>
							</p>
						</div>
						<div id="accordion">
							<div class="card">
								<div class="card-header" id="headingOne">
									<h4 class="mb-0">
										<button class="list-group-item-text" data-toggle="collapse" data-target="#collapseOne" aria-expanded="true" aria-controls="collapseOne">
											Proteins and Complexes
										</button>
									</h4>
								</div>

								<div id="collapseOne" class="collapse" aria-labelledby="headingOne" data-parent="#accordion">
									<div class="card-body">
										<ul>
											<li>
												Structures
												<p>Genome-scale models (GEMs) are used for computational studies of metabolism, protein synthesis, and transcriptional
													regulation based on a specific organism. Recently, structural information has been mapped to those models, which
													are now called genome-scale models with protein structures (GEM-PROs) and provide a new view linking systems
													and structural biology. This pipeline has expanded knowledge of biochemical network properties and can help
													make predictions on whole-cell phenotypes. This comprehensive information can only be visualized in two dimensions,
													so it is necessary to create a pipeline that generates three-dimensional models using GEM-PRO output data. One
													way to generate these models is by gathering the data for the system and putting it into a single “recipe” file
													(will be discussed further in the model assembly) and using this recipe to generate a virtual model through
													a computational engine with built-in assembly algorithms.
												</p>
											</li>
											<li>
												Orientations and Localizations
												<p>For membrane proteins, orientation data was provided by the Orientations of Proteins in Membranes (OPM) database
													and the tool TMHMM, which uses hidden Markov model (a machine learning algorithm) to predict transmembrane helices
													of membrane proteins. For cytoplasmic proteins and membrane proteins without detailed data, the SWISS-MODEL
													homology-modeling tool was found to be extraordinarily useful to predict morphologies. cellPACK includes these
													orientation constraints when packing species onto points on the generated membrane, described below.
												</p>
											</li>
											<li>
												Stoichiometries
												<p>Stoichiometries of proteomics were primarily based off experimental data from
													<a href="https://www.ncbi.nlm.nih.gov/pmc/articles/PMC4888949/" style="color:#30ad63;">Schmidt et al., 2016</a>.
												</p>
												<p>Complex concentrations or copy numbers of complexes per cell were estimated using mass fractions from the ME
													model. The copy numbers were calculated with the assumption that there is a "limiting" complex and that the
													number of complexes were to be maximized. In other words, the mass fraction gave us the ratio between the complexes
													and we wanted to figure out how to best maximize the number of complexes while maintaining the ratio. However,
													because some of the complexes were severely limiting such that less than 1% of proteins would be in complexes,
													another complex would be used as the "limit" instead. The code for this calculation can be found on Github.
												</p>
											</li>
										</ul>
									</div>
								</div>
							</div>

							<div class="card">
								<div class="card-header" id="headingTwo">
									<h4 class="mb-0">
										<button class="list-group-item-text" data-toggle="collapse" data-target="#collapseTwo" aria-expanded="true" aria-controls="collapseTwo">
											Cell Membranes
										</button>
									</h4>
								</div>

								<div id="collapseTwo" class="collapse" aria-labelledby="headingTwo" data-parent="#accordion">
									<div class="card-body">
										<p>Cell Membranes were made using Blender software. E. coli tomograms (add reference and image here) were first used
											to create a representation of the membranes. The tomograms were processed using IMOD to create a 3D representation
											of the membrane. However, due to the lack of full cell tomograms for E. coli, experimental dimensions (listed
											below) were used to reconstruct an estimated membrane representation. The E. coli cell was assumed to be a volume
											composed of a cylinder with two hemishperes attached at the ends.The model from the tomograms was then shrink
											wrapped onto the cylindrical model to give the overall model a more realistic shape. Additionally, the estimated
											model was able to fill in a lot of the incomplete regions of the tomogram derived membranes.</p>
										<div class="col-sm-4">
											<img src="images/tomogramModel.png" height="200" />
										</div>
										<div class="col-sm-4">
											<img src="images/membranes3.png" height="200" />
										</div>
										<div class="col-sm-4">
											<img src="images/blenderMembrane.png" alt="blenderMembrane" height="200" />
										</div>

										<p>The model is scaled such that 1 Angstrom = 1 cm. The E. coli cell model here is approximately 2.5 micrometers long
											by 1.2 micrometers wide. The general width of the periplasm around the middle of the cell (DPB) is about 30-25
											nm and the widths at the ends, PPC1 and PPC2, are 260nm and 80nm, respectively. The experimental dimensions used
											were referenced from (heinemanna and schmidt paper put here) for E. coli MG1655 in LB media. </p>
										<p> Some helpful links:
											<ul>
												<li>
													<a href="http://www.ebi.ac.uk/pdbe/entry/emdb/EMD-3810/" class="text-info">Tomograms</a>
												</li>
												<li>
													<a href="http://bio3d.colorado.edu/imod/" class="text-info">IMOD</a>
												</li>
												<li>
													<a href="https://www.blender.org/" class="text-info">Blender</a>
												</li>
											</ul>
										</p>
									</div>
								</div>
							</div>

							<div class="card">
								<div class="card-header" id="headingThree">
									<h4 class="mb-0">
										<button class="list-group-item-text" data-toggle="collapse" data-target="#collapseThree" aria-expanded="true" aria-controls="collapseThree">
											DNA Model
										</button>
									</h4>
								</div>

								<div id="collapseThree" class="collapse" aria-labelledby="headingThree" data-parent="#accordion">
									<div class="card-body">
										<div class="col-sm-12">
											<img src="images/chromosome.png" class="col-sm-4 img-responsive">
											<p class="col-md-8"> This model uses a 500 bpb (beads per base pair) representation of the E. coli chromosome. Dimensions are about
												1.78 micrometers by 0.78 micrometers with the origin of representation at midcell. The midcell origin was chosen
												arbitrarily. The DNA chromosome model was generously shared with us by William C. Hacker and Professor Adrian
												H. Elcock. </p>
										</div>
									</div>
								</div>
							</div>

							<div class="card">
								<div class="card-header" id="headingFour">
									<h4 class="mb-0">
										<button class="list-group-item-text" data-toggle="collapse" data-target="#collapseFour" aria-expanded="true" aria-controls="collapseFour">
											Integration of all Elements
										</button>
									</h4>
								</div>

								<div id="collapseFour" class="collapse" aria-labelledby="headingFour" data-parent="#accordion">
									<div class="card-body">
										<p> cellPACK </p>
									</div>
								</div>
							</div>
						</div>

						<h3 class="section-1-title">
							Displaying the Model
						</h3>
						<div id="accordion">
							<div class="card">
								<div class="card-header" id="heading21">
									<h4 class="mb-0">
										<button class="list-group-item-text" data-toggle="collapse" data-target="#collapse21" aria-expanded="true" aria-controls="collapse21">
											Backend Development
										</button>
									</h4>
								</div>
								<div id="collapse21" class="collapse" aria-labelledby="heading21" data-parent="#accordion">
									<div class="card-body">
										<p>The data from the "recipe" described in the previous section was integrated into a
											<a href="www.postgresql.org" style="color:#30ad63">PostgresQL</a> relational database containing four tables: models, genes (the primary key table), reactions, and
											complexes. In addition to organizing the data used by the application, this allows easy trawling of the data using
											the search box on the web application. The search function is handled by
											<code>node-postgres</code>, and the entire website and application are built on Node.js. The pages for this website
											are served by a Node.js dyno running on
											<a href="www.heroku.com" style="color:#30ad63">Heroku</a>, a cloud application platform owned by Salesforce and built on Amazon Web Services. The database is
											also hosted on Heroku through the
											<code>heroku-postgres</code> plugin.</p>
										<center>
											<img src="images/db_diagram.png" style="height: 400px;">
										</center>
									</div>
								</div>
								<div class="card">
									<div class="card-header" id="heading22">
										<h4 class="mb-0">
											<button class="list-group-item-text" data-toggle="collapse" data-target="#collapse22" aria-expanded="true" aria-controls="collapse22">
												Frontend Development
											</button>
										</h4>
									</div>
								</div>
								<div id="collapse22" class="collapse" aria-labelledby="heading22" data-parent="#accordion">
									<div class="card-body">
										<p>The web site was prototyped in
											<a href="www.figma.com" style="color:#30ad63">Figma</a> and is being served via a Node.js server hosted on Heroku. The most significant page, of course, is
											the one on which the application runs. The proteins are to be displayed in
											<a href="nglviewer.org" style="color:#30ad63">NGL Viewer</a>, a WebGL/<a href="threejs.org" style="color:#30ad63">Three.js</a> based protein viewing application used by the
											<a href="www.rcsb.org" style="color:#30ad63">RCSB Protein Data Bank</a> to visualize individual structures at high levels of detail. We have successfully loaded
											an entire cellPACK output into NGL Viewer, and are rendering proteins as small spheres until interacted with to
											aid in load times. Unfortunately, interacting with these proteins has proven a formidable challenge, as NGL is
											extremely precise and the proteins must be clicked exactly. Currently, the website features a stock version of
											NGL Viewer, albeit with some slight add-ons such as a coordinate system, while these roadblocks are being resolved.</p>
									</div>
								</div>
								<br>
								<div class="card">
									<div class="card-header" id="heading23">
										<h4 class="mb-0">
											<button class="list-group-item-text" data-toggle="collapse" data-target="#collapse23" aria-expanded="true" aria-controls="collapse23">
												References
											</button>
										</h4>
									</div>
								</div>
								<div id="collapse23" class="collapse" aria-labelledby="heading23" data-parent="#accordion">
									<div class="card-body">
										<pre>Biasini, M., Bienert, S., Waterhouse, A., Arnold, K., Studer, G., Schmidt, T., . . . Schwede, T. (2014, 04). SWISS-MODEL: Modelling protein tertiary and quaternary structure using evolutionary information. Nucleic Acids Research, 42(W1). doi:10.1093/nar/gku340 
Bordbar, A., Monk, J. M., King, Z. A., &amp; Palsson, B. Ø. (2014, 01). Constraint-based models predict metabolic and associated cellular functions. Nature Reviews Genetics, 15(2), 107-120. doi:10.1038/nrg3643
Brunk, E., Mih, N., Monk, J., Zhang, Z., O’Brien, E. J., Bliven, S. E., . . . Palsson, B. Ø. (2016, 03). Systems biology of the structural proteome. BMC Systems Biology, 10(1). doi:10.1186/s12918-016-0271-6
Bureau of Labor Statistics. (2016, 05). Occupational Employment Statistics. Retrieved December 14, 2017, from http://www.bls.gov/oes/
Carrera, J., &amp; Covert, M. W. (2015, 12). Why Build Whole-Cell Models? Trends in Cell Biology, 25(12), 719-722. doi:10.1016/j.tcb.2015.09.004
Cignoni, P., Callieri, M., Corsini, M., Dellapiane, M., Ganovelli, F., &amp; Ranzuglia, G. (2008). MeshLab: An Open-Source Mesh Processing Tool (V. Scarano, R. De Chiara, &amp; U. Erra, Eds.). ERCIM News, 73, 45-46. doi:10.2312/LocalChapterEvents/ItalChap/ItalianChapConf2008/129-136
Cock, P. J., Antao, T., Chang, J. T., Chapman, B. A., Cox, C. J., Dalke, A., . . . Hoon, M. J. (2009, 03). Biopython: Freely available Python tools for computational molecular biology and bioinformatics. Bioinformatics, 25(11), 1422-1423. doi:10.1093/bioinformatics/btp163
Cusick, M. E. (2005, 10). Interactome: Gateway into systems biology. Human Molecular Genetics, 14(Suppl_2). doi:10.1093/hmg/ddi335
Delgado, L., Martínez, G., López-Iglesias, C., &amp; Mercadé, E. (2015, 03). Cryo-electron tomography of plunge-frozen whole bacteria and vitreous sections to analyze the recently described bacterial cytoplasmic structure, the Stack. Journal of Structural Biology, 189(3), 220-229. doi:10.1016/j.jsb.2015.01.008 
Goodsell, D. (2016, 02). Cellular Landscapes in Watercolor. Journal of Biocommunication, 40(1). doi:10.5210/jbc.v40i1.6627
Goodsell, D. S., Autin, L., &amp; Olson, A. J. (2018, 01). Lattice Models of Bacterial Nucleoids. The Journal of Physical Chemistry B. doi:10.1021/acs.jpcb.7b11770
Hacker, W. C., Li, S., Elcock, A. H.  (2017, 06). Features of genomic organization in a nucleotide-resolution molecular model of the Escherichia coli chromosome. Nucleic Acid Research, 45(13), 7541-7554. doi: 10.1093/nar/gkx541
Im, W., Liang, J., Olson, A., Zhou, H., Vajda, S., &amp; Vakser, I. A. (2016, 07). Challenges in structural approaches to cell modeling. Journal of Molecular Biology, 428(15), 2943-2964. doi:10.1016/j.jmb.2016.05.024
Johnson, G. T., Autin, L., Al-Alusi, M., Goodsell, D. S., Sanner, M. F., &amp; Olson, A. J. (2014, 12). CellPACK: A virtual mesoscope to model and visualize structural systems biology. Nature Methods, 12(1), 85-91. doi:10.1038/nmeth.3204
Keskin, O., Tuncbag, N., &amp; Gursoy, A. (2016, 04). Predicting Protein–Protein Interactions from the Molecular to the Proteome Level. Chemical Reviews, 116(8), 4884-4909. doi:10.1021/acs.chemrev.5b00683
Klein, T., Autin, L., Kozlikova, B., Goodsell, D. S., Olson, A., Groller, M. E., &amp; Viola, I. (2017). Instant Construction and Visualization of Crowded Biological Environments. IEEE Transactions on Visualization and Computer Graphics, 1-1. doi:10.1109/tvcg.2017.2744258
Kremer, J. R., Mastronarde, D. N., &amp; Mcintosh, J. (1996, 01). Computer Visualization of Three-Dimensional Image Data Using IMOD. Journal of Structural Biology, 116(1), 71-76. doi:10.1006/jsbi.1996.0013
Krogh, Anders, et al. Predicting Transmembrane Protein Topology with a Hidden Markov Model: Application to Complete Genomes. Journal of Molecular Biology, vol. 305, no. 3, 2001, pp. 567–580., doi:10.1006/jmbi.2000.4315. 
Le Gros, M. A., Mcdermott, G., &amp; Larabell, C. A. (2005, 10). X-ray tomography of whole cells. Current Opinion in Structural Biology, 15(5), 593-600. doi:10.1016/j.sbi.2005.08.008
Lee, R., Karr, J. R., &amp; Covert, M. W. (2013). WholeCellViz: Data visualization for whole-cell models. BMC Bioinformatics, 14(1), 253. doi:10.1186/1471-2105-14-253
Lomize, M. A., Lomize, A. L., Pogozheva, I. D., &amp; Mosberg, H. I. (2006, 01). OPM: Orientations of Proteins in Membranes database. Bioinformatics, 22(5), 623-625. doi:10.1093/bioinformatics/btk023
Lučić, V., Rigort, A., &amp; Baumeister, W. (2013, 08). Cryo-electron tomography: The challenge of doing structural biology in situ. The Journal of Cell Biology, 202(3), 407-419. doi:10.1083/jcb.201304193
Macklin, D. N., Ruggero, N. A., &amp; Covert, M. W. (2014, 08). The future of whole-cell modeling. Current Opinion in Biotechnology, 28, 111-115. doi:10.1016/j.copbio.2014.01.012
Mih, N., Brunk, E., Chen, K., Catoiu, E., Sastry, A., Kavvas, E., . . . Palsson, B. Ø. (2018, 02). ssbio: A Python framework for structural systems biology. Bioinformatics. doi:10.1093/bioinformatics/bty077
Milne, J. L., &amp; Subramaniam, S. (2009, 08). Cryo-electron tomography of bacteria: Progress, challenges and future prospects. Nature Reviews Microbiology, 7(9), 666-675. doi:10.1038/nrmicro2183
O’Brien, E. J., Lerman, J. A., Chang, R. L., Hyduke, D. R., &amp; Palsson, B. Ø. (2014, 04). Genome-scale models of metabolism and gene expression extend and refine growth phenotype prediction. Molecular Systems Biology, 9(1), 693-693. doi:10.1038/msb.2013.52
Ortiz, J. O., Brandt, F., Matias, V. R., Sennels, L., Rappsilber, J., Scheres, S. H., . . . Baumeister, W. (2010, 08). Structure of hibernating ribosomes studied by cryoelectron tomography in vitro and in situ. The Journal of Cell Biology, 190(4), 613-621. doi:10.1083/jcb.201005007
Purcell, O., Jain, B., Karr, J. R., Covert, M. W., &amp; Lu, T. K. (2013, 06). Towards a whole-cell modeling approach for synthetic biology. Chaos: An Interdisciplinary Journal of Nonlinear Science, 23(2), 025112. doi:10.1063/1.4811182
Rajagopala, S. V., Sikorski, P., Kumar, A., Mosca, R., Vlasblom, J., Arnold, R., . . . Uetz, P. (2014, 02). The binary protein-protein interaction landscape of Escherichia coli. Nature Biotechnology, 32(3), 285-290. doi:10.1038/nbt.2831
Rose, A. S., &amp; Hildebrand, P. W. (2015, 04). NGL Viewer: A web application for molecular visualization. Nucleic Acids Research, 43(W1). doi:10.1093/nar/gkv402
Santarella-Mellwig, R., Pruggnaller, S., Roos, N., Mattaj, I. W., &amp; Devos, D. P. (2013, 05). Three-Dimensional Reconstruction of Bacteria with a Complex Endomembrane System. PLoS Biology, 11(5). doi:10.1371/journal.pbio.1001565
Schmidt, A., Kochanowski, K., Vedelaar, S., Ahrné, E., Volkmer, B., Callipo, L., . . . Heinemann, M. (2015). The quantitative and condition-dependent Escherichia coli proteome. Nature Biotechnology, 34(1), 104-110. doi:10.1038/nbt.3418
Spirin, V., &amp; Mirny, L. A. (2003, 09). Protein complexes and functional modules in molecular networks. Proceedings of the National Academy of Sciences, 100(21), 12123-12128. doi:10.1073/pnas.2032324100
Szklarczyk, D., Morris, J. H., Cook, H., Kuhn, M., Wyder, S., Simonovic, M., . . . Von Mering, C. (2016, 10). The STRING database in 2017: Quality-controlled protein–protein association networks, made broadly accessible. Nucleic Acids Research, 45(D1). doi:10.1093/nar/gkw937
Webber, J. (2012). A programmatic introduction to Neo4j. Proceedings of the 3rd Annual Conference on Systems, Programming, and Applications: Software for Humanity - SPLASH '12. doi:10.1145/2384716.2384777
Yuan, S., Chan, H. S., &amp; Hu, Z. (2017, 01). Using PyMOL as a platform for computational drug design. Wiley Interdisciplinary Reviews: Computational Molecular Science, 7(2). doi:10.1002/wcms.1298
										</pre>
									</div>
								</div>
							</div>
						</div>
					</div>
				</div>
			</div>
		</div>



		<footer id="vr-footer">
			<div class="container">
				<div class="row">
					<div class="footer-links">
						<div class="footer-link">
							<ol>
								<li>
									<a href="contact">contact us</a>
								</li>
								<li>
									<a href="privacyPolicy">privacy policy</a>
								</li>
								<li>
									<a href="termsOfUse">terms of use</a>
								</li>
								<li>
									<a href="https://github.com/rohangro/vizrecon">github</a>
								</li>
							</ol>
						</div>
					</div>
					<div class="footer-copyright">
						<span>© 2017 Systems Biology Research Group at the University of California San Diego. See downloads page for licensing information.
							<br> The content of this site does not necessarily reflect the views of the Regents of the University of California.<br><small>Page leads: Winnie Shi and Rohan Grover</small></span>
					</div>
					<div class="ask-spacer-sm"></div>
				</div>
			</div>
		</footer>
	</div>


	<!-- jQuery -->
	<script src="js/jquery.min.js"></script>
	<!-- jQuery Easing -->
	<script src="js/jquery.easing.1.3.js"></script>
	<!-- Bootstrap -->
	<script src="js/bootstrap.min.js"></script>
	<!-- Waypoints -->
	<script src="js/jquery.waypoints.min.js"></script>
	<!-- Magnific Popup -->
	<script src="js/jquery.magnific-popup.min.js"></script>
	<!-- Owl Carousel -->
	<script src="js/owl.carousel.min.js"></script>
	<!-- toCount -->
	<script src="js/jquery.countTo.js"></script>
	<!-- wow -->
	<script src="js/wow.min.js"></script>
	<!-- Main JS -->
	<script src="js/main.js"></script>
</body>

</html>